module/zfs/zfs_vnops.c

```diff
@@ -3563,6 +3652,17 @@ zfs_rename(struct inode *sdip, char *snm, struct inode *tdip, char *tnm,
 	}
 
 	/*
+	 * If we are using project inheritance, we only allow renames
+	 * into our tree when the project IDs are the same; otherwise
+	 * tree quota mechanism would be circumvented.
```

Comment:
Cool.  In that case can you reword this comment, since this restriction (of PROJINHERIT being also force-inherited) applies all the time, not just for the (non-existent) tree quota.  It seems like the PROJINHERIT flag is potentially useful even in the absence of tree quota, since it allows you to force a directory and everything below it to be assigned to the same project.